test/impersonate_test.exs

```diff
@@ -1,8 +1,38 @@
 defmodule ImpersonateTest do
   use ExUnit.Case
   doctest Impersonate
+  require Impersonate.GenServer
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  describe "assert_receive_call/1" do
+    test "matches the pattern" do
+      parent = self()
+      Task.async(fn -> GenServer.call(parent, {:hello, :world}) end)
+      Impersonate.GenServer.assert_receive_call {:hello, :world}
+    end
+  end
+
+  describe "assert_received_call/1" do
+    test "matches the pattern" do
+      parent = self()
+      Task.async(fn -> GenServer.call(parent, {:hello, :world}) end)
```

Comment:
It would be good hygiene to handle the async reply here, imo. The `Task.await` call might help you to avoid the `Process.sleep` below.